addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -336,7 +336,14 @@ var SnippetEditor = Widget.extend({
             return;
         }
         this.willDestroyEditors = true;
-        await this.toggleTargetVisibility(!this.$target.hasClass('o_snippet_invisible'));
+        // Toggle the visibility to "true" of all the elements that do not have
+        // the "o_snippet_invisible" class as well as the elements that have a
+        // conditional visibility. Indeed, we want to see those elements the
+        // next time we enter in "Edit" mode. Elements for which
+        // "toggleTargetVisibility" will be called with "false" as arguments are
+        // for example, the popup snippet or a hidden header or footer.
+        await this.toggleTargetVisibility(!this.$target.hasClass("o_snippet_invisible") ||
+            this.$target[0].dataset.visibility === "conditional");
```

Comment:
I think this is wrong, we shoud hide the target when its `visibility === "conditional"` no ?

For the moment:
- Drop a block
- Set a conditional visibility like visible only for logged out users
- Save
=> The block flickers at the beginning of the page :/

Reply:
@Guillaume-gdi , not sure to understand your comment here. Let's remember that targets that have `visibility === "conditional"` also have the `o_snippet_invisible` class. So if I understood correctly, what you would like to do is 
```js
await this.toggleTargetVisibility(!this.$target.hasClass("o_snippet_invisible"));
```
and you hide the target when its `visibility === "conditional"`. The problem of doing so is that this will happen:
- Drop a block
- Set the visibility to "Conditionally" -> you want to see the snippet on your website.
- Save
=> The snippet is not visible on the page.
Indeed, if at the "save", you call `this.toggleTargetVisibility(false)` for the snippet. What you will do is add the `o_conditional_hidden` class to the snippet. Because the elements that have this class are hidden on the page, the snippet is hidden but you actually want to see it. This class should not be present on the elements that have a conditional visibility out of the edit mode as the rule that define their visibility or not on the page is defined thanks to css rule at the `DOMContentLoaded` (see `inject_dom.js` file).

By doing
```js
await this.toggleTargetVisibility(!this.$target.hasClass("o_snippet_invisible") ||
            this.$target[0].dataset.visibility === "conditional");
```
`this.toggleTargetVisibility(true)` is called for snippet that have a conditional visibility. By doing so, the `o_conditional_hidden` class is removed (as the visibility or not of the snippet out of the edit mode is defined by the rule defined by the user) and the `data-invisible` attribute is removed. Note that this last attribute is responsible of the crossed eye icon on the "Invisible elements" panel in edit mode. By removing it at the "save",  you are sure that the eye will be in the "visible" mode the next time the user enters in Edit mode.

Reply:
Note that the flickering can be explained like this: 
- At Save request remove the `o_conditional_hidden` class to the snippet so it is visible.
- At DOM contend loaded the css rule is applied and the snippet is hidden.
→ Due to the time between those two operations, a flickering effect appears.

Note that the reverse phenomena exists in V15 (see video):
- Add the `o_conditional_hidden` class so the snippet is hidden.
- At the DOM contend loaded, the css rule is applied and the snippet is shown.
→ Due to the time between those two operations, a flickering effect appears.
However, maybe is it better like this as things are shown little by little rather than hiding little by little (this could lead to unwanted reveal of a special promotion for ex?)

[Screencast from 05-12-2023 09:40:32 PM.webm](https://github.com/odoo/odoo/assets/114914971/27b44c76-f560-4648-a34a-77ddbbec84d2)

Reply:
Okay, maybe this line is correct but the bug I described is still there and need to be fixed. A conditionnaly visible block shouldn't be visible and then disapear but should always be invisible and then appear (if the condition is met of course).

Video of the issue:
[Runbot R&D.webm](https://github.com/odoo/odoo/assets/78849981/98191102-381d-456d-9522-57422ea5633c)

Reply:
@Guillaume-gdi , this is now fixed :+1: